src/game/Mario.js

```diff
@@ -775,6 +796,7 @@ const update_mario_geometry_inputs = (m) => {
     m.waterLevel = -20000.0 //find_water_level(m->pos[0], m->pos[2]);
 
     if (m.floor) {
+        if (!m.floor.normal) m.floor.normal = {z: 0, x: 0};
```

Comment:
broken pls fix

Reply:
So, I added throw error here if m.floor.normal is null,  and so far that error has never gone off.   Any ideas?  Do you know what conditions caused problems here?

Reply:
not sure, might have been fixed with your sin/cos changes!